src/qibo/validation/gate_set_tomography.py

```diff
@@ -0,0 +1,1273 @@
+import os
+import time
+
+import numpy as np
+
+from qibo import gates
+from qibo.backends import GlobalBackend
+from qibo.models import Circuit
+
+
+def prepare_1qb_state_new(k):
+    nqubits = 1
+
+    k_to_gates_1qb = {
+        0: (),
+        1: (gates.X(0),),
+        2: (gates.H(0),),
+        3: (gates.H(0), gates.S(0)),
+    }
+
+    circ = Circuit(nqubits, density_matrix=True)
+    for gate in k_to_gates_1qb[k]:
+        circ.add(gate)
+    return circ
+
+
+def prepare_2qb_state_new(k):
+    nqubits = 2
+
+    k_to_gates_2qb = {
+        0: (),
+        1: (gates.X(1),),
+        2: (gates.H(1),),
+        3: (gates.H(1), gates.S(1)),
+        4: (gates.X(0),),
+        5: (
+            gates.X(0),
+            gates.X(1),
+        ),
+        6: (
+            gates.X(0),
+            gates.H(1),
+        ),
+        7: (gates.X(0), gates.H(1), gates.S(1)),
+        8: (gates.H(0),),
+        9: (
+            gates.H(0),
+            gates.X(1),
+        ),
+        10: (
+            gates.H(0),
+            gates.H(1),
+        ),
+        11: (gates.H(0), gates.H(1), gates.S(1)),
+        12: (
+            gates.H(0),
+            gates.S(0),
+        ),
+        13: (
+            gates.H(0),
+            gates.S(0),
+            gates.X(1),
+        ),
+        14: (
+            gates.H(0),
+            gates.S(0),
+            gates.H(1),
+        ),
+        15: (gates.H(0), gates.S(0), gates.H(1), gates.S(1)),
+    }
+
+    circ = Circuit(nqubits, density_matrix=True)
+    for gate in k_to_gates_2qb[k]:
+        circ.add(gate)
+    return circ
+
+
+def measure_1qb_state_new(j, circ, noise_model=None):
+    nqubits = 1
+
+    j_to_measurements_1qb = {
+        0: (gates.M(0, basis=gates.Z),),
```

Comment:
To get the expectation values of Pauli operators {I, X, Y, Z}, we measure them in their different bases. To measure in the X basis using the default computational basis, we just apply Hadamard gate to rotate it and then do a measurement in the computational basis. We do likewise for Y and Z. For I, we just measure in the computational basis but use +1 for both P(0) and P(1) to get the expectation value for I. Computing <I> is a necessary step in GST. 

Since there's already a function ```basis=gates.Y``` to measure in the Y basis, we can use that without manually applying the gates. But there's no ```basis=gates.I``` so I just measure in Z.  Likewise below.